Fewer unwanted events tutorial (#4496)
PostHog/posthog.com: contents/tutorials/fewer-unwanted-events.md

---
title: How to capture fewer unwanted events
sidebar: Docs
showTitle: true
author: ['ian-vanagas']
date: 2022-10-20
featuredImage: ../images/tutorials/banners/fewer-unwanted-events.png

> topics: ['configuration', 'apps']

Reviewer: ```suggestion
topics: ['events', 'apps']
```

## Using feature flags

If you’re worried that a specific area of your product generating too many events, you can put those events behind feature flags. You can use feature flags instead of changing code to remove events or turn off autocapture. Here are two ways to do so.

First, you can turn off autocapture with a feature flag when the PostHog library loads. Toggling off this feature flag can lower event flow within PostHog without changing the code.

```js
posthog.init(
  '<ph_project_api_key>',
  { 
    api_host: '<ph_instance_address>',
    loaded: function (posthog) {
      if (posthog.isFeatureEnabled('disable-autocapture')) {
        posthog.config.autocapture = false;
      }
    }
  }
)
```

You can also put events in key areas behind feature flags and turn them off if you reach your limit.

```js
if (!posthog.isFeatureEnabled('disable-event-capture')) {
	posthog.capture('event');
}
```

Feature flags work well as “kill switches” in situations where you want fewer events. You can collect events from a specific area and then turn the event capture off when you’ve got the data and insights you’re looking for.

## Drop events based on property app

> PostHog has apps that enable you to modify the events flowing into your instance. We can use them to capture fewer events. 

Reviewer 2: ```suggestion
PostHog has apps that enable you to modify the events flowing into your instance and which you can use to control how many events you capture. 
```

> The first app is the “[Drop events based on property](https://github.com/PostHog/drop-events-on-property-plugin)” app. You can use it to drop events that match a specified property. This is useful for [privacy-focused](/tutorials/property-filter) teams or teams who want to capture fewer events. 

Author: Property filter is different than Drop events based on property. 

Property filter sets property to null. Drop events based on property checks for property and drops event if it matches key value pair. In this way, it is similar to https://github.com/plibither8/posthog-filter-out-plugin

I've added Filter Out to the tutorial so maybe I just remove Drop events based on property and we can drop it from the listing.

> The first app is the “[Drop events based on property](https://github.com/PostHog/drop-events-on-property-plugin)” app. You can use it to drop events that match a specified property. This is useful for [privacy-focused](/tutorials/property-filter) teams or teams who want to capture fewer events. 

Reviewer 2: OK, so it sounds like we have an app which isn't available in the app store and needs to be though. I'll flag it with support hero (@mariusandra) and when it's available I'll create a new listing for it.

To set up this app, search for “Drop Events Based On Property” in Apps, click the blue gear, add the key and value (optional) of the event you want to drop, click save, and activate the toggle. As an example, if I want to drop events related to a specific page, I can set the property key to `$pathname` and the property value to `/about`

![Drop events based on property app](../images/tutorials/fewer-unwanted-events/drop-events.png)

Doing this drops any events where the property `$pathname` is `/about`. This is useful if certain pages on your site create a lot of events, but aren’t useful to you. Other uses include dropping events from a specific OS, browser, device type, location, user (distinct ID), and more.

## Downsampling app

The second app you can use to capture fewer events is the [Downsampler](/docs/apps/downsampling) app. It reduces the number of events your instance will ingest by a percentage. 

To configure it, search for the “Downsampling Plugin” in Apps, click the blue gear, pick a percentage of events you want to keep, and click the toggle to activate.

![Downsampler app](../images/tutorials/fewer-unwanted-events/downsampler.png)

> The problem with downsampling (compared to the other methods covered) is that you have less control over event ingestion. The app drops a random selection of events and you must guess what percentage you need. A percentage of a large number can still be a large number, which might not solve your problem. On the other hand, you could also be downsampling when you don’t need to. Downsampling is an option, but we'd recommend trying the other first.

Reviewer 3: the really specific problem here is that it doesn't drop events for a set % of _users_, it's just overall

this means it'll cause some really weird looking "impossible" behavior like users doing step 1 and 3 of a funnel, but not stop 2

Reviewer 3: I'd explicitly not recommend this app for this reason

Reviewer 3: and for really low volume things imagine "customer paid over $100K" or whatever, if you get unlucky, it'll drop the one or two really important things that happened that you care about, depending on the nature of your product

Author: Yes, this was partially the inspiration for this piece, people dropping important events or misconfiguring apps to drop events they didn't want to drop.

I've added more warnings around the downsampler app to let users know the potential consequences of using it.
